### PR TITLE
Ubuntu/xenial

### PR DESCRIPTION
**Pull request by blackboxsw**
## Proposed Commit Message

This branch will not be squash merged, but reviewed and landed on the command line via `build-and-push` tooling by the autho once approved.

## Additional Context
This branch was created to sync a new-upstream-snapshot into Xenial for release.  quilt patches needed updating to properly merge ds-identify changes which are applied to retain original Xenial ds-identify behavior.

## Test Steps

This branch was created with the following:
```bash

git clone git@github.com:canonical/cloud-init.git
cd cloud-init
git checkout upstream/ubuntu/xenial -b ubuntu/xenial


cat > di.patch <<EOF
diff -urN tools/ds-identify tools/ds-identify.new 
--- tools/ds-identify	2020-11-25 13:06:26.851398596 -0700
+++ tools/ds-identify.new	2020-11-25 13:06:53.927809332 -0700
@@ -94,8 +94,8 @@
 
 DI_BLKID_EXPORT_OUT=""
 DI_GEOM_LABEL_STATUS_OUT=""
-DI_DEFAULT_POLICY="search,found=all,maybe=all,notfound=${DI_DISABLED}"
-DI_DEFAULT_POLICY_NO_DMI="search,found=all,maybe=all,notfound=${DI_ENABLED}"
+DI_DEFAULT_POLICY="report,found=all,maybe=all,notfound=${DI_ENABLED}"
+DI_DEFAULT_POLICY_NO_DMI="report,found=all,maybe=all,notfound=${DI_ENABLED}"
 DI_DMI_CHASSIS_ASSET_TAG=""
 DI_DMI_PRODUCT_NAME=""
 DI_DMI_SYS_VENDOR=""

EOF

new-upstream-snapshot 47f4229ebcef9f83df8b549bb869a2dbf6dff17c

# then quilt fix ds-identify patch for xenial
(refresh-fix) ... quilt push -f
(refresh-fix) ... patch -p0 < di.patch
(refresh-fix) ... quilt refresh
(refresh-fix) ... exit 0

# Add SRU bug number 1905599 in place of SRU_BUG_NUMBER in debian/changelog
# follow new-upstream-snapshot closing comments to set debian/changelog and commit message etc


# fix debian/changelog leaked LP: # references in commit summaries
sed -i 's/LP: #1900837/#1900837/' debian/changelog
sed -i 's/LP: #1886531/#1886531/' debian/changelog
git commit -am 'redact leaked LP bugs in commit summaries'

```
## Checklist:

 - [X] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [x] I have tested building my package with `build-package`
 - [x] I have have run `sbuild` against my dsc

### CODE REVIEW

File: debian/patches/ds-identify-behavior-xenial.patch [hunk 1]
```diff
@@ -10,18 +10,18 @@ Bug-ubuntu: http://bugs.launchpad.net/bugs/1660385
...
- DI_MAIN=${DI_MAIN:-main}
+@@ -94,8 +94,8 @@ DI_MAIN=${DI_MAIN:-main}
  
- DI_BLKID_OUTPUT=""
+ DI_BLKID_EXPORT_OUT=""
+ DI_GEOM_LABEL_STATUS_OUT=""
 -DI_DEFAULT_POLICY="search,found=all,maybe=all,notfound=${DI_DISABLED}"
 -DI_DEFAULT_POLICY_NO_DMI="search,found=all,maybe=all,notfound=${DI_ENABLED}"
-+DI_DEFAULT_POLICY="report,found=all,maybe=all,notfound=${DI_ENABLED}"
++DI_DEFAULT_POLICY="report,found=all,maybe=all,notfound=${DI_DISABLED}"
```
Comment (OddBloke) [inline debian/patches/ds-identify-behavior-xenial.patch:19]: This doesn't look right to me; the diff we needed to fix daily was ed5471f1cce33ef17a13321cb6436a2b4dc9973d:

```diff
--- a/debian/patches/ds-identify-behavior-xenial.patch
+++ b/debian/patches/ds-identify-behavior-xenial.patch
@@ -10,10 +10,10 @@ Bug-ubuntu: http://bugs.launchpad.net/bugs/1660385
 
 --- a/tools/ds-identify
 +++ b/tools/ds-identify
-@@ -93,8 +93,8 @@ _DI_LOGGED=""
- DI_MAIN=${DI_MAIN:-main}
+@@ -94,8 +94,8 @@
  
- DI_BLKID_OUTPUT=""
+ DI_BLKID_EXPORT_OUT=""
+ DI_GEOM_LABEL_STATUS_OUT=""
 -DI_DEFAULT_POLICY="search,found=all,maybe=all,notfound=${DI_DISABLED}"
 -DI_DEFAULT_POLICY_NO_DMI="search,found=all,maybe=all,notfound=${DI_ENABLED}"
 +DI_DEFAULT_POLICY="report,found=all,maybe=all,notfound=${DI_ENABLED}"
@@ -21,7 +21,7 @@ Bug-ubuntu: http://bugs.launchpad.net/bugs/1660385
  DI_DMI_CHASSIS_ASSET_TAG=""
  DI_DMI_PRODUCT_NAME=""
  DI_DMI_SYS_VENDOR=""
-@@ -131,7 +131,7 @@ DI_ON_FOUND=""
+@@ -132,7 +132,7 @@
  DI_ON_MAYBE=""
  DI_ON_NOTFOUND=""
```

Comment (blackboxsw) [reply debian/patches/ds-identify-behavior-xenial.patch:19]: @OddBloke I can't seem to find that commitish you mentioned anywhere.  Did you mean the commit in master cf6c36a18de233190996f7c2dde1e04908bad50a? which introduced part of that change to ds-identify?

Comment (OddBloke) [reply debian/patches/ds-identify-behavior-xenial.patch:19]: I meant https://github.com/canonical/cloud-init/commit/dd17691cc24a3b69b67b56078e147bca3e283ae6; evidently I pushed and then somehow changed my local history!

Comment (blackboxsw) [reply debian/patches/ds-identify-behavior-xenial.patch]: > This doesn't look right to me; the diff we needed to fix daily was ed5471f1cce33ef17a13321cb6436a2b4dc9973d
Thanks a lot for catching that. my diff now against upstream/ubuntu/xenial is now


```diff
ubuntu/xenial)$ git diff origin/ubuntu/xenial  debian/patches/ds-identify-behavior-xenial.patch 
diff --git a/debian/patches/ds-identify-behavior-xenial.patch b/debian/patches/ds-identify-behavior-xenial.patch
index ba7639ab..25cc3813 100644
--- a/debian/patches/ds-identify-behavior-xenial.patch
+++ b/debian/patches/ds-identify-behavior-xenial.patch
@@ -10,10 +10,10 @@ Bug-ubuntu: http://bugs.launchpad.net/bugs/1660385
 
 --- a/tools/ds-identify
 +++ b/tools/ds-identify
-@@ -93,8 +93,8 @@ _DI_LOGGED=""
- DI_MAIN=${DI_MAIN:-main}
+@@ -94,8 +94,8 @@ DI_MAIN=${DI_MAIN:-main}
  
- DI_BLKID_OUTPUT=""
+ DI_BLKID_EXPORT_OUT=""
+ DI_GEOM_LABEL_STATUS_OUT=""
 -DI_DEFAULT_POLICY="search,found=all,maybe=all,notfound=${DI_DISABLED}"
 -DI_DEFAULT_POLICY_NO_DMI="search,found=all,maybe=all,notfound=${DI_ENABLED}"
 +DI_DEFAULT_POLICY="report,found=all,maybe=all,notfound=${DI_ENABLED}"
@@ -21,7 +21,7 @@ Bug-ubuntu: http://bugs.launchpad.net/bugs/1660385
  DI_DMI_CHASSIS_ASSET_TAG=""
  DI_DMI_PRODUCT_NAME=""
  DI_DMI_SYS_VENDOR=""
-@@ -131,7 +131,7 @@ DI_ON_FOUND=""
+@@ -132,7 +132,7 @@ DI_ON_FOUND=""
  DI_ON_MAYBE=""
  DI_ON_NOTFOUND=""
```

**Review by OddBloke — APPROVED** [review]
This matches what I see when I follow the process, and successfully builds. Thanks!

**Pull request merged**